File: zemacs-term/tests/fixtures/injection/dao.py
import re
from sqlalchemy import text


def run(cursor):
    # execute method -> sql
    cursor.execute("SELECT exec_col FROM exec_tbl WHERE id = 1")

    # sqlalchemy text() -> sql
    stmt = text("UPDATE text_tbl SET x = 1")

    # triple-quoted content auto-detect -> sql
    big = """
    SELECT auto_py_a, auto_py_b
    FROM auto_py
    WHERE id = 2
    """

    # regex via re.compile -> regex
    pat = re.compile("[0-9]+regexpy")

    # plain string -> NOT injected
    label = "select an item from the dropdown list"
    return stmt, big, pat, label

File: zemacs-term/tests/fixtures/injection/Dao.kt
fun run(db: android.database.sqlite.SQLiteDatabase) {
    // method call -> sql
    val c = db.rawQuery("SELECT kt_col FROM kt_tbl WHERE id = 1")

    // multiline content auto-detect -> sql
    val q = """
        UPDATE kt_auto SET a = 1 WHERE id = 2
    """

    // plain string -> NOT injected
    val label = "pick a theme from the kotlin settings"
}
